cmd/triggerrun/cmd/root.go

```diff
@@ -50,27 +54,28 @@ var (
 		Long:  "tkn-trigger will allow users easily test out the Trigger config.",
 		RunE:  rootRun,
 	}
-
+	action      string
 	triggerFile string
 	httpPath    string
 )
 
 func init() {
+	rootCmd.Flags().StringVarP(&action, "show or create", "a", "", "it's to show or create resources")
 	rootCmd.Flags().StringVarP(&triggerFile, "triggerFile", "t", "", "Path to trigger yaml file")
 	rootCmd.Flags().StringVarP(&httpPath, "httpPath", "r", "", "Path to body event")
 	rootCmd.PersistentFlags().StringVarP(&kubeconfig, "kubeconfig", "k", "", "absolute path to the kubeconfig file")
 }
 
 func rootRun(cmd *cobra.Command, args []string) error {
-	err := trigger(triggerFile, httpPath)
+	err := trigger(triggerFile, httpPath, action)
 	if err != nil {
 		return fmt.Errorf("fail to call trigger: %v", err)
 	}
 
 	return nil
 }
 
-func trigger(triggerFile, httpPath string) error {
+func trigger(triggerFile, httpPath, action string) error {
```

Comment:
the trigger function now contains quite a bit of logic...we should add some tests for it.